src/jdk.jlink/share/classes/jdk/tools/jlink/internal/Platform.java

```diff
@@ -24,112 +24,59 @@
  */
 package jdk.tools.jlink.internal;
 
+import jdk.internal.util.Architecture;
+import jdk.internal.util.OperatingSystem;
+
 import java.util.Locale;
 
 /**
- * Supported platforms
+ * Supported OperatingSystem and Architecture.
  */
 public record Platform(OperatingSystem os, Architecture arch) {
 
-    public enum OperatingSystem {
-        WINDOWS,
-        LINUX,
-        MACOS,
-        AIX,
-        UNKNOWN;
-    }
-
-    public enum Architecture {
-        X86,
-        x64,
-        ARM,
-        AARCH64,
-        UNKNOWN;
-    }
-
-    public static final Platform UNKNOWN = new Platform(OperatingSystem.UNKNOWN, Architecture.UNKNOWN);
-
     /*
      * Returns the {@code Platform} based on the platformString of the form <operating system>-<arch>.
+     * @throws IllegalArgumentException if the delimiter is missing or either OS or
+     * architecture is not known
      */
     public static Platform parsePlatform(String platformString) {
         String osName;
         String archName;
         int index = platformString.indexOf("-");
         if (index < 0) {
-            osName = platformString;
-            archName = "UNKNOWN";
-        } else {
-            osName = platformString.substring(0, index);
-            archName = platformString.substring(index + 1);
+            throw new IllegalArgumentException("platformString missing delimiter: " + platformString);
         }
-        OperatingSystem os;
-        try {
-            os = OperatingSystem.valueOf(osName.toUpperCase(Locale.ENGLISH));
-        } catch (IllegalArgumentException e) {
-            os = OperatingSystem.UNKNOWN;
-        }
-        Architecture arch = toArch(archName);
+        osName = platformString.substring(0, index);
+        OperatingSystem os = OperatingSystem.valueOf(osName.toUpperCase(Locale.ROOT));
+
+        archName = platformString.substring(index + 1);
+        // Alias architecture "amd64" to "X64"
+        archName = archName.replace("amd64", "X64");
+        Architecture arch = Architecture.valueOf(archName.toUpperCase(Locale.ROOT));
+
         return new Platform(os, arch);
     }
 
     /**
-     * @return true is it's a 64-bit platform
+     * {@return true is it's a 64-bit platform}
+     * All platform exception X86 are 64-bit platforms.
      */
     public boolean is64Bit() {
-        return (arch() == Platform.Architecture.x64 ||
-                arch() == Platform.Architecture.AARCH64);
+        return arch != Architecture.X86;
```

Comment:
Have you considered having an instance `Architecture::is64bit` method in lieu of the current static method?

Reply:
Possibly, if a need subsequently arises. 
The only current use is in CDSPlugin and it is testing if the current/runtime platform is 64-bit so it can create the non-oop archive. The `Architecture.is64bit()` provides that.  The current `Platform.is64Bit()` method is not needed.